sos/report/plugins/networking.py

```diff
@@ -141,16 +142,10 @@ def setup(self):
                 "ethtool --show-eee " + eth
             ], tags=eth)
 
-            # skip EEPROM collection for 'bnx2x' NICs as this command
-            # can pause the NIC and is not production safe.
-            bnx_output = {
-                "cmd": "ethtool -i %s" % eth,
-                "output": "bnx2x"
-            }
-            bnx_pred = SoSPredicate(self,
-                                    cmd_outputs=bnx_output,
-                                    required={'cmd_outputs': 'none'})
-            self.add_cmd_output("ethtool -e %s" % eth, pred=bnx_pred)
+            # skip EEPROM collection by default, as it might hang or
+            # negatively impact the system on some device types
+            if self.get_option("eepromdump"):
+                self.add_cmd_output("ethtool -e %s" % eth)
```

Comment:
Should we be enabling this for devices that are already "known bad", even when the option is used? I.e. bnx2, possibly tg3 and others?

(genuine question - I could argue it either way...)

Reply:
I would personally leave that up to the operator to decide that. If the default is to skip the eeprom dumps, then that seems pretty safe and sane to me.

Reply:
We could add a warning message here perhaps? That way sysadmins still have a chance to abort the collection.

Something like this maybe?

`WARNING: collecting an eeprom dump is known to cause certain NIC drivers (e.g. bnx2x/tg3) to interrupt device operation`

Reply:
> I would personally leave that up to the operator to decide that. If the default is to skip the eeprom dumps, then that seems pretty safe and sane to me.

The problem is that nothing currently describes these actions as potentially dangerous (which we know from experience they are). Neither the kernel ioctl documentation, nor the ethtool docs mention any possible side effects from the eeprom dump collection. Given that we are now aware of these problems I think we should communicate that to the user - a warning message is an option, but I think the option description should also flag that using this plugin option has the potential for very serious side effects.

Reply:
iow "leave it to the operator" assumes that the operator is properly informed and understands what they are consenting to. I don't think that's the case currently.